src/System.ServiceModel.Syndication/src/System/ServiceModel/Syndication/XmlSyndicationContent.cs

```diff
@@ -168,14 +141,16 @@ protected override void WriteContentsTo(XmlWriter writer)
         {
             if (writer == null)
             {
-                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("writer");
+                throw new ArgumentNullException(nameof(writer));
             }
-            if (_extension != null)
+
+            if (Extension != null)
             {
-                _extension.WriteTo(writer);
+                Extension.WriteTo(writer);
             }
-            else if (_contentBuffer != null)
+            else
             {
+                Debug.Assert(_contentBuffer != null, "contentBuffer cannot be null");
```

Comment:
_contentBuffer can be null. Put the null check back and remove the Debug.Assert

Reply:
I disagree - other code can prove this: https://github.com/dotnet/corefx/blob/master/src/System.ServiceModel.Syndication/src/System/ServiceModel/Syndication/XmlSyndicationContent.cs#L122-L142

Essentially either the buffer is null or the extension is null. They can't both be simultaneously null

Reply:
You're right, I had missed the mutually exclusive nature of this with Extension.

---
In reply to: [192364573](https://github.com/dotnet/corefx/pull/29882#discussion_r192364573) [](ancestors = 192364573)